lib/factory_bot/registry.rb

```diff
@@ -1,4 +1,20 @@
 module FactoryBot
+  # In Ruby <= 2.5 it is necessary to monkey patch
+  # the key error class because it does not expose
+  # receiver and key attributes.
+  # See https://bugs.ruby-lang.org/issues/14313
+  module KeyErrorExtension
+    attr_accessor :receiver, :key
```

Comment:
I was under the impression there methods were always available, these just weren't using instance variables before. Do we need to redefine them?

Reply:
You're right, patching initialize should be enough. Dropping ...

Reply:
Turns out, the attributes exist but are not exposed means we need at least ``attr_reader``.